reaper.go

```diff
@@ -87,12 +96,39 @@ func NewReaper(ctx context.Context, sessionID string, provider ReaperProvider, r
 	return reaper, nil
 }
 
+func extractDockerHost(ctx context.Context) (dockerHostPath string) {
+	if dockerHostPath = os.Getenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE"); dockerHostPath != "" {
+		return dockerHostPath
+	}
+
+	dockerHostPath = "/var/run/docker.sock"
+
+	var hostRawURL string
+	if h, ok := ctx.Value(dockerHostContextKey).(string); !ok || h == "" {
```

Comment:
Does `dockerHostContextKey` need to be a new type? Could it be just a string? Not a blocker, trying to understand its usage

Reply:
See the [docs](https://pkg.go.dev/context#WithValue) regarding why this is a type :blush: there's also a blog entry on go.dev I think but I can't find it right now. Does this cover your question ?

Reply:
>Users of WithValue should define their own types for keys.

LGTM